crates/ruff/src/rules/ruff/rules/unreachable.rs

```diff
@@ -0,0 +1,964 @@
+    /// `range(10)`.
+    Iterator(&'stmt Expr),
+    Match {
+        /// `match $subject`.
+        subject: &'stmt Expr,
+        /// `case $case`, include pattern, guard, etc.
+        case: &'stmt MatchCase,
+    },
+}
+
+impl<'stmt> Ranged for Condition<'stmt> {
+    fn range(&self) -> TextRange {
+        match self {
+            Condition::Test(expr) | Condition::Iterator(expr) => expr.range(),
+            // The case of the match statement, without the body.
+            Condition::Match { subject: _, case } => TextRange::new(
+                case.start(),
+                case.guard
+                    .as_ref()
+                    .map_or(case.pattern.end(), |guard| guard.end()),
+            ),
+        }
+    }
+}
+
+/// Index into [`BasicBlocks::blocks`].
+type BlockIndex = usize;
+
+impl<'stmt> BasicBlock<'stmt> {
+    /// A sentinel block indicating an empty termination block.
+    const EMPTY: BasicBlock<'static> = BasicBlock {
+        stmts: &[],
+        next: NextBlock::Terminate,
+    };
+
+    /// A sentinel block indicating an exception was raised.
+    const EXCEPTION: BasicBlock<'static> = BasicBlock {
+        stmts: &[Stmt::Return(StmtReturn {
+            range: TextRange::new(TextSize::new(0), TextSize::new(0)),
+            value: None,
+        })],
+        next: NextBlock::Terminate,
+    };
+
+    /// Return true if the block is a sentinel or fake block.
+    fn is_sentinel(&self) -> bool {
+        self.is_empty() || self.is_exception()
+    }
+
+    /// Returns an empty block that terminates.
+    fn is_empty(&self) -> bool {
+        matches!(self.next, NextBlock::Terminate) && self.stmts.is_empty()
+    }
+
+    /// Returns true if `self` an [`BasicBlock::EXCEPTION`].
+    fn is_exception(&self) -> bool {
+        matches!(self.next, NextBlock::Terminate) && BasicBlock::EXCEPTION.stmts == self.stmts
+    }
+}
+
+/// Creates basic blocks from `stmts` and appends them to `blocks`.
+fn create_blocks<'stmt>(
+    blocks: &mut Vec<BasicBlock<'stmt>>,
+    stmts: &'stmt [Stmt],
+    mut after: Option<BlockIndex>,
+) {
+    // We process the statements in reverse so that we can always point to the
+    // next block (as that should always be processed).
+    let mut stmts_iter = stmts.iter().enumerate().rev().peekable();
+    while let Some((i, stmt)) = stmts_iter.next() {
+        let next = match stmt {
+            // Statements that continue to the next statement after execution.
+            Stmt::FunctionDef(_)
+            | Stmt::AsyncFunctionDef(_)
+            | Stmt::Import(_)
+            | Stmt::ImportFrom(_)
+            | Stmt::ClassDef(_)
+            | Stmt::Global(_)
+            | Stmt::Nonlocal(_)
+            | Stmt::Delete(_)
+            | Stmt::Assign(_)
+            | Stmt::AugAssign(_)
+            | Stmt::AnnAssign(_)
+            | Stmt::Expr(_)
+            | Stmt::Break(_)
+            | Stmt::Continue(_) // NOTE: the next branch gets fixed up in `change_next_block`.
```

Comment:
Nit: Does this comment apply for `Break` and `Continue`. If so, I think it would make sense to move these two instruction into their own match arm even when they both have the same body.

Reply:
No only for `Continue`, break should stop looping and continue to the statement after the loop. `Break` however should return to the loop statement (creating the loop).